### PR TITLE
Stop exabgp before stopping and removing PTF container in cleanup

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205

#### What is the motivation for this PR?
There have been some efforts to fix the test server stuck issue. However, the test server cleanup scenario was missed. When we run testbed-cli.sh cleanup-vmhost, the cleanup.sh script will be eventually called. This script has steps to remove the PTF containers without stopping the exabgp service in them in advance. This also could result in test server stuck.

The previous efforts trying to work around this issue:
https://github.com/sonic-net/sonic-mgmt/pull/6131
https://github.com/sonic-net/sonic-mgmt/pull/6191
https://github.com/sonic-net/sonic-mgmt/pull/6241
#### How did you do it?
1. Stop exabgp services before stopping all running docker containers in cleanup.sh.j2 file.
2. Get all ptf containers in current server, and use docker exec to get all exabgp services of ptf containers, then use supervisorctl stop to stop all exabgp services.

#### How did you verify/test it?
Run `testbed-cli.sh cleanup-vmhost`

### CODE REVIEW

File: ansible/roles/vm_set/templates/cleanup.sh.j2 [hunk 1]
```diff
@@ -17,6 +17,16 @@ rm -f {{ root_path }}/disks/*
...
 test -z "$(ovs-vsctl list-br)" || ovs-vsctl list-br | xargs -I % ovs-vsctl del-br %
 
+# stop exabgp process before stopping ptf container
+test -z "$(docker ps --format "table {{.Names}}" | grep ptf)"
+ptf_container_name=$(docker ps --format "table {{.Names}}" | grep ptf)
+for ptf in  $ptf_container_name;
+do
+  test -z "$(docker exec $ptf supervisorctl status | grep exabgp | awk '{print $1}')"
+  exabgp_process_name="$(docker exec $ptf supervisorctl status | grep exabgp | awk '{print $1}')"
+  docker exec $ptf supervisorctl stop $exabgp_process_name
```
Comment (wangxin) [inline ansible/roles/vm_set/templates/cleanup.sh.j2:27]: There could be multiple exabgp services running in PTF container.
I am not sure if that scenario can be handled by the code here.

Comment (xwjiang-ms) [reply ansible/roles/vm_set/templates/cleanup.sh.j2:27]: supervisorctl stop can stop multiple processes, so multiple exabgp services could be stopped

**Comment by wangxin** [comment]
Prepare testbed hit this issue while running `testbed-cli.sh add-topo`:
```
TASK [vm_set : Install cleanup script] *****************************************
Monday 24 October 2022  07:49:33 +0000 (0:00:00.368)       0:00:20.586 ******** 
fatal: [STR-ACS-VSERV-01]: FAILED! => {"changed": false, "msg": "AnsibleError: template error while templating string: unexpected '.'. String: #!/bin/bash\n\nset -e\n\n# stop rynning VMs\ntest -z \"$(virsh list --state-running --name)\" || virsh list --state-running --name | xargs -I % virsh destroy %\n\n# stop paused VMs\ntest -z \"$(virsh list --state-paused --name)\" || virsh list --state-running --name | xargs -I % virsh destroy %\n\n# undefine all VMs\ntest -z \"$(virsh list --all --name)\" || virsh list --all --name | xargs -I % virsh undefine %\n\n# remove all VM disks\nrm -f {{ root_path }}/disks/*\n\n# remove all ovs bridges\ntest -z \"$(ovs-vsctl list-br)\" || ovs-vsctl list-br | xargs -I % ovs-vsctl del-br %\n\n# stop exabgp process before stopping ptf container\ntest -z \"$(docker ps --format \"table {{.Names}}\" | grep ptf)\"\nptf_container_name=$(docker ps --format \"table {{.Names}}\" | grep ptf)\nfor ptf in  $ptf_container_name;\ndo\n  test -z \"$(docker exec $ptf supervisorctl status | grep exabgp | awk '{print $1}')\"\n  exabgp_process_name=\"$(docker exec $ptf supervisorctl status | grep exabgp | awk '{print $1}')\"\n  docker exec $ptf supervisorctl stop $exabgp_process_name\ndone\n\n# stop all running docker containers\ntest -z \"$(docker ps -q)\" || docker stop $(docker ps -q)\n\n# remove all docker containers\ntest -z \"$(docker ps -a -q)\" || docker rm -f $(docker ps -a -q)\n\n# remove all docker images\ntest -z \"$(docker images -q)\" || docker rmi -f $(docker images -q|uniq)\n"}

```

**Comment by xwjiang-ms** [comment]
> Prepare testbed hit this issue while running `testbed-cli.sh add-topo`:
> 
> ```
> TASK [vm_set : Install cleanup script] *****************************************
> Monday 24 October 2022  07:49:33 +0000 (0:00:00.368)       0:00:20.586 ******** 
> fatal: [STR-ACS-VSERV-01]: FAILED! => {"changed": false, "msg": "AnsibleError: template error while templating string: unexpected '.'. String: #!/bin/bash\n\nset -e\n\n# stop rynning VMs\ntest -z \"$(virsh list --state-running --name)\" || virsh list --state-running --name | xargs -I % virsh destroy %\n\n# stop paused VMs\ntest -z \"$(virsh list --state-paused --name)\" || virsh list --state-running --name | xargs -I % virsh destroy %\n\n# undefine all VMs\ntest -z \"$(virsh list --all --name)\" || virsh list --all --name | xargs -I % virsh undefine %\n\n# remove all VM disks\nrm -f {{ root_path }}/disks/*\n\n# remove all ovs bridges\ntest -z \"$(ovs-vsctl list-br)\" || ovs-vsctl list-br | xargs -I % ovs-vsctl del-br %\n\n# stop exabgp process before stopping ptf container\ntest -z \"$(docker ps --format \"table {{.Names}}\" | grep ptf)\"\nptf_container_name=$(docker ps --format \"table {{.Names}}\" | grep ptf)\nfor ptf in  $ptf_container_name;\ndo\n  test -z \"$(docker exec $ptf supervisorctl status | grep exabgp | awk '{print $1}')\"\n  exabgp_process_name=\"$(docker exec $ptf supervisorctl status | grep exabgp | awk '{print $1}')\"\n  docker exec $ptf supervisorctl stop $exabgp_process_name\ndone\n\n# stop all running docker containers\ntest -z \"$(docker ps -q)\" || docker stop $(docker ps -q)\n\n# remove all docker containers\ntest -z \"$(docker ps -a -q)\" || docker rm -f $(docker ps -a -q)\n\n# remove all docker images\ntest -z \"$(docker images -q)\" || docker rmi -f $(docker images -q|uniq)\n"}
> ```

Fixed in newest commit, that because jinja2 can't template "." in table {{.Names}}, and I used awk '{print $NF}' instead

**Review by wangxin — APPROVED** [review]

**Pull request merged**